src/status_im/ui/screens/wallet/send/sheets.cljs

```diff
@@ -0,0 +1,77 @@
+(ns status-im.ui.screens.wallet.send.sheets
+  (:require-macros [status-im.utils.views :refer [defview letsubs] :as views])
+  (:require [re-frame.core :as re-frame]
+            [status-im.i18n :as i18n]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.list-item.views :as list-item]
+            [status-im.ui.components.chat-icon.screen :as chat-icon]
+            [status-im.ui.components.list.views :as list]
+            [status-im.ui.screens.wallet.accounts.views :as wallet.accounts]
+            [status-im.utils.utils :as utils.utils]))
+
+(views/defview assets [address]
+  (views/letsubs [{:keys [tokens]} [:wallet/visible-assets-with-values address]
+                  currency [:wallet/currency]]
+    [list/flat-list
+     {:data      tokens
+      :key-fn    (comp str :symbol)
+      :render-fn (wallet.accounts/render-asset
+                  (:code currency)
+                  #(do
+                     (re-frame/dispatch [:bottom-sheet/hide-sheet])
+                     (re-frame/dispatch [:set-in [:wallet/prepare-transaction :symbol] (:symbol %)])))}]))
+
+(defn render-account [account]
+  [list-item/list-item
+   {:icon     [chat-icon/custom-icon-view-list (:name account) (:color account)]
+    :title    (:name account)
+    :on-press #(do
+                 (re-frame/dispatch [:bottom-sheet/hide-sheet])
+                 (re-frame/dispatch [:set-in [:wallet/prepare-transaction :from] account]))}])
+
+(views/defview accounts-list []
+  (views/letsubs [{:keys [accounts]} [:multiaccount]]
+    [list/flat-list {:data      accounts
+                     :key-fn    :address
+                     :render-fn render-account}]))
+
+(defn- request-camera-permissions []
+  (let [options {:handler :wallet.send/qr-scanner-result
+                 :cancel-handler :wallet.send/qr-scanner-cancel}]
+    (re-frame/dispatch
+     [:request-permissions
+      {:permissions [:camera]
+       :on-allowed
+       #(do
+          (re-frame/dispatch [:bottom-sheet/hide-sheet])
```

Comment:
is code that got moved? could you create an event to avoid the triple dispatch?

Reply:
will fix to one event